Gemfile.lock

```diff
@@ -1,57 +1,57 @@
 GEM
   remote: https://rubygems.org/
   specs:
-    actionpack (5.2.1)
-      actionview (= 5.2.1)
-      activesupport (= 5.2.1)
+    actionpack (5.2.2.1)
+      actionview (= 5.2.2.1)
+      activesupport (= 5.2.2.1)
       rack (~> 2.0)
       rack-test (>= 0.6.3)
       rails-dom-testing (~> 2.0)
       rails-html-sanitizer (~> 1.0, >= 1.0.2)
-    actionview (5.2.1)
-      activesupport (= 5.2.1)
+    actionview (5.2.2.1)
```

Comment:
### [actionview](http://rubyonrails.org)

**Patch** version upgrade :chart_with_upwards_trend::small_blue_diamond: 5.2.1 → 5.2.2.1

[_[change-log](https://github.com/rails/rails/blob/v5.2.2.1/actionview/CHANGELOG.md), [source-code](https://github.com/rails/rails/tree/v5.2.2.1/actionview)_]

:tada: Patched vulnerabilities:

- [CVE-2019-5419](https://nvd.nist.gov/vuln/detail/CVE-2019-5419)
  **Denial of Service Vulnerability in Action View**

  
  URL: https://groups.google.com/forum/#!topic/rubyonrails-security/GN7w9fFAQeI

  There is a potential denial of service vulnerability in actionview. This vulnerability has been assigned the CVE identifier CVE-2019-5419.  Impact ------ Specially crafted accept headers can cause the Action View template location code to consume 100% CPU, causing the server unable to process requests.  This impacts all Rails applications that render views.  All users running an affected release should either upgrade or use one of the workarounds immediately.  Workarounds ----------- This vulnerability can be mitigated by wrapping `render` calls with `respond_to` blocks.  For example, the following example is vulnerable:  ``` class UserController < ApplicationController   def index     render "index"   end end ```  But the following code is not vulnerable:  ``` class UserController < ApplicationController   def index     respond_to |format|       format.html { render "index" }     end   end end ```  Implicit rendering is impacted, so this code is vulnerable:  ``` class UserController < ApplicationController   def index   end end ```  But can be changed this this:  ``` class UserController < ApplicationController   def index     respond_to |format|       format.html { render "index" }     end   end end ```  Alternatively to specifying the format, the following monkey patch can be applied in an initializer:  ``` $ cat config/initializers/formats_filter.rb # frozen_string_literal: true  ActionDispatch::Request.prepend(Module.new do   def formats     super().select do |format|       format.symbol || format.ref == "*/*"     end   end end) ```  Credits  -------  Thanks to John Hawthorn <john@hawthorn.email> of GitHub


- [CVE-2019-5418](https://nvd.nist.gov/vuln/detail/CVE-2019-5418)
  **File Content Disclosure in Action View**

  
  URL: https://groups.google.com/forum/#!topic/rubyonrails-security/pFRKI96Sm8Q

  There is a possible file content disclosure vulnerability in Action View. This vulnerability has been assigned the CVE identifier CVE-2019-5418.  Versions Affected:  All. Not affected:       None. Fixed Versions:     6.0.0.beta3, 5.2.2.1, 5.1.6.2, 5.0.7.2, 4.2.11.1  Impact ------ There is a possible file content disclosure vulnerability in Action View. Specially crafted accept headers in combination with calls to `render file:` can cause arbitrary files on the target server to be rendered, disclosing the file contents.  The impact is limited to calls to `render` which render file contents without a specified accept format.  Impacted code in a controller looks something like this:  ``` class UserController < ApplicationController   def index     render file: "#{Rails.root}/some/file"   end end ```  Rendering templates as opposed to files is not impacted by this vulnerability.  All users running an affected release should either upgrade or use one of the workarounds immediately.  Releases -------- The 6.0.0.beta3, 5.2.2.1, 5.1.6.2, 5.0.7.2, and 4.2.11.1 releases are available at the normal locations.  Workarounds ----------- This vulnerability can be mitigated by specifying a format for file rendering, like this:  ``` class UserController < ApplicationController   def index     render file: "#{Rails.root}/some/file", formats: [:html]   end end ```  In summary, impacted calls to `render` look like this:  ``` render file: "#{Rails.root}/some/file" ```  The vulnerability can be mitigated by changing to this:  ``` render file: "#{Rails.root}/some/file", formats: [:html] ```  Other calls to `render` are not impacted.  Alternatively, the following monkey patch can be applied in an initializer:  ``` $ cat config/initializers/formats_filter.rb # frozen_string_literal: true  ActionDispatch::Request.prepend(Module.new do   def formats     super().select do |format|       format.symbol || format.ref == "*/*"     end   end end) ```  Credits ------- Thanks to John Hawthorn <john@hawthorn.email> of GitHub






<details>
<summary>Commits</summary>

A change of **126** commits. See the full changes on [the compare page](https://github.com/rails/rails/compare/v5.2.1...v5.2.2.1).

These are the first 10 commits:
- (cf90cba) [backport use_authenticated_cookie_encryption doc fixes [ci s…](https://github.com/rails/rails/commit/cf90cba2becaa971b1b9140bbe83bc4b101d8187)
- (10679c2) [Merge pull request #33325 from Edouard-chin/ec-deprecate-cla…](https://github.com/rails/rails/commit/10679c2cbd25550f6625dcac0d9f813e8e91b2ab)
- (dd11336) [Avoid extra scoping when using `Relation#update`](https://github.com/rails/rails/commit/dd1133699f7266e9659c3624de1c77fcd43a4691)
- (ede8bc0) [Merge pull request #33489 from Atul9/remote-ip](https://github.com/rails/rails/commit/ede8bc0c77016bb93b8a2957fdb941c3b2bb0e59)
- (9897cf5) [Merge pull request #33488 from znz/avoid-to-post-twice](https://github.com/rails/rails/commit/9897cf55e2609e1e67ec0cb50ea7f2cff95fcc1d)
- (0d2cc93) [Merge pull request #33452 from thetizzo/rails_32_to_40_doc_u…](https://github.com/rails/rails/commit/0d2cc930acd1ebe5f0c1cb7729088ebbde2c2452)
- (067913f) [Merge pull request #33507 from jackc/patch-1](https://github.com/rails/rails/commit/067913f95d38f6acd11223f6b35f9386c31c94ca)
- (ccf90fb) [Fix uploading Tempfiles to Azure Storage](https://github.com/rails/rails/commit/ccf90fb2e0a4717e63bad87e87409facc3d3b2da)
- (250fc41) [Tempfile.open requires a name in Ruby 2.3](https://github.com/rails/rails/commit/250fc41df47a44545a68f82429f21c118a224a21)
- (c32462a) [Merge branch '5-2-1' into 5-2-stable](https://github.com/rails/rails/commit/c32462a0db5005961f2f83973e44d90c6ad86a5f)


</details>